### ISSUE TITLE
Deferred: 421 Too many concurrent SMTP connections

### ISSUE THREAD

**Issue by jgonggrijp** [spamgourmet/spamgourmet#41]
I ran into an old thread on the BBS, which seems significant enought that it might still be an issue with spamgourmet:

https://bbs.spamgourmet.com/viewtopic.php?t=1406

There are some other threads which report the same issue. As diagnosed by the thread participants, it appears that MailGun might overload spamgourmet with concurrent SMTP connections, causing spamgourmet to start rejecting them. Other email services appear less likely to make this rejection.

**Comment by maxgraser** [comment]
I have posted an update to the thread.

As written I've also had this issue but it seems it has now been resolved as I'm starting to receive Emails from one of the services again (don't have any other services with the same issue).
For documentation the issue:
- started 02. August. 2019 (last email received from no-reply@s.backerkit.com)
- ended 19. July. 2022 (first email received from no-reply@s.backerkit.com)

**Comment by mikedlr** [comment]
I don't think this issue is resolved as such. A bit over a week ago I had the following

> Your message hasn't been delivered yet
> 
> The message you sent to [word.user@somesgdomain.tld](mailto:word.user@somesgdomain.tld) hasn't been delivered yet due to: Recipient server unavailable or busy.
> 
> We'll let you know if we can't deliver it successfully.
> 
> 
> Further Information
> 
> Too many concurrent SMTP connections from this IP address; please try again later.

(see also [recent issues thread](https://bbs.spamgourmet.com/viewtopic.php?f=4&t=1889&sid=23ec0d50d8971a31c744342db97ee9a6) on BBS) 

I think this is inevitable with some of the big mail delivery providers like mailgun and similar (there are quite a few). In order to solve this we just have to up the limit either with configuration or, more likely, by having multiple incoming servers and IP addresses.

**Comment by mikedlr** [comment]
N.B. I should add that this was from a recruitment company and using _mimecast_ and so likely slightly "spammy" but it was for mails that I definitely wanted so it's definitely a vaild spamgourmet problem.

**Comment by fredericmora** [comment]
How many connections do we need to ensure they don't run out? We can try
raising the limit a bit.

**Comment by mikedlr** [comment]
I don't think there's any way to set a finite particular limit. There's quite a bit of consolidation going on and one particular provider could be deliver a huge proportion of legitimate spamgourmet email and trying to do that from a small number of servers. 

What I think we want to know is whether this is a pretty legitimate / important limit for spammers? I'm guessing that most of the time that it gets hit that *is* the sign of an actual spammer??

I guess we should just set up some monitoring and reporting - which IP addresses are we ending up blocking mails from / how often / for what reason.

**Comment by lwcorp** [comment]
This is very troubling. My hosting for example hasn't been able to send me e-mails for years because of it (they pinpointed it to this).
They said "We have no similar reports from any other of our clients."

**Comment by fluo75** [comment]
This is issue is arising again these days. This is a real issue :(
